src/prism.c

```diff
@@ -21939,7 +21945,8 @@ pm_parser_init(pm_parser_t *parser, const uint8_t *source, size_t size, const pm
             size_t length = (size_t) ((newline != NULL ? newline : parser->end) - cursor);
             if (length > 2 && cursor[0] == '#' && cursor[1] == '!') {
                 if (parser->newline_list.size == 1) {
```

Comment:
@kddnewton I don't think this case can every be hit.

`parser->newline_list` starts out with `size = 1` (with `offset[0] = 0`). The `pm_newline_list_append` on line 21934 above will cause `parser->newline_list.size` here to always be >2

Reply:
In this case, can you add some tests and verify that we have the correct behavior in these cases?

Reply:
Do you know what Ruby code even triggers this case? I couldn't figure it out, so I had just forced it to reach this path by editing the `if` predicates.

Reply:
Done.